Content.Server/Physics/Controllers/MoverController.cs

```diff
@@ -63,48 +145,68 @@ protected override bool CanSound()
+        // We use _seenMovers here as well as in UpdateMoverStatus—this means we
+        // cannot have any events get fired while we use it in this while loop.
+        _seenMovers.Clear();
+        _moversToUpdate.Clear();
 
-        // Need to order mob movement so that movers don't run before their relays.
-        while (inputQueryEnumerator.MoveNext(out var uid, out var mover))
+        // Don't use EntityQueryEnumerator because admin ghosts have to move on
+        // paused maps. Pausing movers is handled via ActiveInputMoverComponent.
+        var inputQueryEnumerator = AllEntityQuery<ActiveInputMoverComponent, InputMoverComponent>();
+        while (inputQueryEnumerator.MoveNext(out var uid, out var activeComp, out var moverComp))
         {
-            InsertMover((uid, mover));
+            _seenRelayMovers.Clear(); // O(1) if already empty
+            QueueRelaySources(activeComp.RelayedFrom);
+
+            // If it's already inserted, that's fine—that means it'll still be
+            // handled before its child movers
+            AddMover(uid, moverComp);
         }
 
-        foreach (var mover in _movers)
+        ActiveMoverGauge.Set(_moversToUpdate.Count);
+
+        foreach (var ent in _moversToUpdate)
         {
-            HandleMobMovement(mover, frameTime);
+            HandleMobMovement(ent, frameTime);
         }
 
-        ActiveMoverGauge.Set(_movers.Count);
-
         HandleShuttleMovement(frameTime);
+        return;
+
+        // When we insert a chain of relay sources we have to flip its ordering
+        // It's going to be extremely uncommon for a relay chain to be more than
+        // one entity so we just recurse as needed.
+        void QueueRelaySources(EntityUid? next)
+        {
+            // We only care if it's still a mover
+            if (!_activeQuery.TryComp(next, out var nextActive)
+                || !MoverQuery.TryComp(next, out var nextMover)
+                || !_seenRelayMovers.Add(next.Value))
+                return;
+
+            // While it is (as of writing) currently true that this recursion
+            // should always terminate due to RelayedFrom always being written
+            // in a way that tracks if it's made a loop, we still take the extra
+            // memory (and small time cost) of making sure via _seenRelayMovers.
+            QueueRelaySources(nextActive.RelayedFrom);
+            AddMover(next.Value, nextMover);
+        }
+
+        // Track inserts so we have ~ O(1) inserts without duplicates. Hopefully
+        // it doesn't matter that both _seenMovers and _moversToUpdate are never
+        // trimmed? They should be pretty memory light anyway, and in general
+        // it'll be rare for there to be a decrease in movers.
+        void AddMover(EntityUid uid, InputMoverComponent comp)
+        {
+            if (_seenMovers.Contains(uid))
+                return;
+            _moversToUpdate.Add((uid, comp));
+            _seenMovers.Add(uid);
```

Comment:
```suggestion
            if (!_seenMovers.Add(uid))
                return;
            _moversToUpdate.Add((uid, comp));     
```
I think this should be equivalent and slightly more efficient.